### PR TITLE
Add builder methods to simplify adding common kotlin.jvm annotations

### LINKED ISSUE #190: Nice support for @JvmField etc.
Ergonomics.

### PR DESCRIPTION
**Pull request by Egorand**
Closes #190

### CODE REVIEW

**Comment by JakeWharton** [comment]
Let's do these as extensions in a `jvm` subpackage? I don't see any reason they need to live directly on the type since they depend entirely on the public API.

Plus it helps a future multiplatform story.

**Comment by JakeWharton** [comment]
(This is how the stdlib does a solid 90% of its functionality)

**Comment by Egorand** [comment]
Agreed!

**Comment by JakeWharton** [comment]
Looks great otherwise. Vast simplification of common cases.

**Comment by Egorand** [comment]
Any other annotations you feel are worth adding?

**Comment by JakeWharton** [comment]
I mean I would probably do all of the platform-specific annotations in
platform packages. We can just start with the JVM ones though.

**Comment by Egorand** [comment]
I can add all kotlin.jvm in this PR and then follow up with other platforms

File: src/test/java/com/squareup/kotlinpoet/JvmAnnotationsTest.kt [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        .addFunction(FunSpec.builder("foo")
+            .throws(ClassName("com.squareup.tacos", "IllegalTacoException"))
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import kotlin.jvm.Throws
+      |
+      |@Throws(IllegalTacoException::class)
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/JvmAnnotationsTest.kt]: 🌮👮‍♀️

File: src/test/java/com/squareup/kotlinpoet/JvmAnnotationsTest.kt [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import kotlin.Int
+      |import kotlin.String
+      |import kotlin.jvm.JvmOverloads
+      |
+      |class Taco @JvmOverloads constructor(bar: Int, baz: String = "baz")
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/JvmAnnotationsTest.kt]: Nice example

File: src/test/java/com/squareup/kotlinpoet/jvm/JvmAnnotationsTest.kt [hunk 3]
```diff
@@ -0,0 +1,821 @@
...
+            .build())
+        .build()
+    assertThat(file.toString()).isEqualTo("""
+      |package com.squareup.tacos
+      |
+      |import kotlin.Int
+      |import kotlin.collections.List
+      |import kotlin.jvm.JvmSuppressWildcards
+      |
+      |fun foo(a: List<@JvmSuppressWildcards kotlin.Int>) {
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/jvm/JvmAnnotationsTest.kt]: #346

**Comment by Egorand** [comment]
@JakeWharton PTAL. I left out `@JvmSynthetic` and `@PurelyImplements` - I doubt anyone's gonna ever use these. 

In this case, extension functions will be almost unusable for Java users, is this a problem?

File: src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt [hunk 4]
```diff
@@ -573,27 +574,18 @@ class TypeSpecTest {
...
-                .addMember("%T::class", IOException::class.asClassName())
-                .build())
+            .throws(IOException::class)
             .build())
         .addFunction(FunSpec.builder("nativeThrow")
             .addModifiers(KModifier.EXTERNAL)
-            .addAnnotation(AnnotationSpec.builder(Throws::class)
-                .addMember("%T::class", IOException::class.asClassName())
-                .build())
+            .throws(IOException::class)
```
Comment (swankjesse) [inline src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt]: much nicer

**Pull request merged**